recipes/date/all/conanfile.py

```diff
@@ -40,46 +48,55 @@ def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
         if self.settings.os in ["iOS", "tvOS", "watchOS", "Android"]:
-            self.options.use_system_tz_db = True
+            self.options.tz_db = "system"
 
     def configure(self):
         if self.options.shared or self.options.header_only:
             self.options.rm_safe("fPIC")
         if self.options.header_only:
             del self.options.shared
+            del self.options.tz_db
+            del self.options.use_tz_db_in_dot
             self.package_type = "header-library"
 
     def layout(self):
         cmake_layout(self, src_folder="src")
 
-    def requirements(self):
-        if not self.options.header_only and not self.options.use_system_tz_db:
-            self.requires("libcurl/[>=7.78 <9]")
-
     def package_id(self):
         if self.info.options.header_only:
             self.info.clear()
 
     def validate(self):
         if self.settings.compiler.get_safe("cppstd"):
             check_min_cppstd(self, 11)
+        if self.options.get_safe("tz_db") == "system" and self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("Using system tz database is not supported on Windows")
+        if self.options.get_safe("tz_db") != "download" and self.options.get_safe("use_tz_db_in_dot"):
+            raise ConanInvalidConfiguration("Option 'use_tz_db_in_dot'=True cannot be used with 'tz_db' != 'download'")
+
+    def requirements(self):
+        if self.version == "2.4.1" or self.options.get_safe("tz_db") == "download":
+            self.requires("libcurl/[>=7.78 <9]")
 
     def source(self):
         get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     def generate(self):
         tc = CMakeToolchain(self)
-        tc.variables["ENABLE_DATE_TESTING"] = False
-        tc.variables["USE_SYSTEM_TZ_DB"] = self.options.use_system_tz_db
-        tc.variables["USE_TZ_DB_IN_DOT"] = self.options.use_tz_db_in_dot
-        tc.variables["BUILD_TZ_LIB"] = not self.options.header_only
-        # workaround for gcc 7 and clang 5 not having string_view
-        if Version(self.version) >= "3.0.0" and \
-            ((self.settings.compiler == "gcc" and Version(self.settings.compiler.version) <= "7.0") or \
-             (self.settings.compiler == "clang" and Version(self.settings.compiler.version) <= "5.0")):
-            tc.cache_variables["DISABLE_STRING_VIEW"] = True
+        tc.cache_variables["ENABLE_DATE_TESTING"] = False
```

Comment:
```suggestion
        tc.cache_variables["ENABLE_DATE_TESTING"] = False
        if Version(self.version) < 3:
            tc.cache_variables["CMAKE_POLICY_VERSION_MINIMUM"] = "3.5"
```

Since Date 3.0.4 they are requiring CMake 3.7. Older versions need 3.2 

https://github.com/HowardHinnant/date/commit/fc4cf092f9674f2670fb9177edcdee870399b829